src/TestFramework/Extension.Shared/TestContext.cs

```diff
@@ -21,6 +22,11 @@ public abstract class TestContext
         /// </summary>
         public abstract IDictionary Properties { get; }
 
+        /// <summary>
+        /// Gets or sets the cancellation token source that is cancelled when test timesout or when cancelled the test will be aborted
+        /// </summary>
+        public CancellationTokenSource CancellationTokenSource { get; protected set; }
```

Comment:
Add this change for Desktop TestContext as well : https://github.com/Microsoft/testfx/blob/master/src/TestFramework/Extension.Desktop/TestContext.cs#L18